### ISSUE TITLE
CSV inport into protege

### ISSUE THREAD

**Issue by JosephAWhite** [protegeproject/protege#1062]
Hello,

This is probably not a new question, but it does not appear to have a solution.

I have several CSV files of terms for a series of OWL classes.  The terms would be "individuals".  There is no obvious way to import these individuals into a Protege class.  There are hundreds of terms, and I DO NOT want to manually enter individuals.  

It would seem to me that building some sort of importer for flat files into Protege would have been built already.  Does a mechanism for directly importing a CSV of terms into Protege exist?   If not, why not?  I am exceptionally frustrated and about to drop Protege altogether.  Not having a mechanism like this seems to me to a major oversight.  

I am looking at ROBOT, but its not so easy to use, and does not do what I want it to do, ie import my CSV of terms into an OWL format with a default class.  I can edit the terms once they're loaded into Protege; I just need to get them there first.  

Can anyone suggest a suitable simple mechanism to do this?

Thanks,
Joe White

**Comment by JLouize** [comment]
Hi Joe, 

take a look at https://github.com/protegeproject/cellfie-plugin.

**Comment by mdebellis** [comment]
Yes, Cellfie should do what you want. I find Cellfie works better if I import the CSV into Excel or Google docs spreadsheet first and then use that file. Also, I'm not sure if this is what you had in mind but I think there should be some equivalent to Tools>Create class hierarchy. I.e., you should be able to click on a class and say Tools>Create instances and then just type or paste a long list of Individual names and have them all of them be instances of whatever class you created. I'm surprised they have this for class, object, and data property hierarchies but not for instances. When all you want to do is just create lots of instances (rather than also import values for various properties) then Cellfie is overkill and something simple like that would be very useful. If something like that exists somewhere and I just haven't seen it I would appreciate a pointer to it.
